### ISSUE TITLE
in-process postgres proxy auth errors

### ISSUE THREAD

**Issue by SandyWalsh** [GoogleCloudPlatform/cloudsql-proxy#155]
As others have reported, I'm getting the dreaded `The client is not authorized to make this request., notAuthorized` error.

I have a service account created and credential info available, but I don't know how to pass this into into the proxy when running it in-process (not sidecar). Is this even possible?

It seems I can only give username/password type stuff to the "cloudsqlpostgres" driver. 

Can the in-process proxy use the cred.json file?

**Comment by hfwang** [comment]
Cloud SQL proxy uses google.DefaultClient to setup its credentials

(see https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/b858be3de41449cd1ff3b30f1131f20eda15e825/proxy/proxy/dial.go#L92)

The logic for how DefaultClient gets its credentials is described in FindDefaultCredentials (https://godoc.org/golang.org/x/oauth2/google#FindDefaultCredentials)

TLDR: You can set the GOOGLE_APPLICATION_CREDENTIALS to the path to the credential JSON file.
